### PR TITLE
#167313387 Configure Postgres and Sequelize

### PR DESCRIPTION
**Pull request by habinezadalvan**
**What does this PR do?**

- Configures Authors Haven to use PostgreSQL as DB Engine and Sequelize.

- Configures babel

**Description of Task to be completed?**

- Created the src folder with config, migrations, models, seeds folders

- installed sequelize, pg, pg-hstore

- Created authors-heaven database

- Configured the Postgres

- Created User model and run migrations to conform that tables are being created in authors heaven database

**How should this be manually tested?**

- Clone the repo, run <npm install>, create "authors-heaven" db, create .env file and include credentials as specified in the config.js inside config folder, the run <npm run migration>
**Screenshot**
Screen Shot 2019-07-30 at 22 02 21

**What are the relevant pivotal tracker stories?**
[#167313387](https://www.pivotaltracker.com/story/show/167313387)

@mikeanguandia @minega25 @salviosage @nshutijonathan @nkalyesubula @Mnickii

### CODE REVIEW

**Review by minega25 — CHANGES_REQUESTED** [review]
In `src/config/config.js` I would suggest the database name for the test environment to match one used in travis which is `authorsheaven_test`.

File: src/models/user.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+module.exports = (sequelize, DataTypes) => {
+  const User = sequelize.define('User', {
+    firstname: DataTypes.STRING,
+    lastname: DataTypes.STRING,
+    username: DataTypes.STRING,
```
Comment (nshutijonathan) [inline src/models/user.js]: Would you please allow null as false from here? yesterday at the sync patrick. minega suggested about that

**Review by nshutijonathan — APPROVED** [review]

**Review by salviosage — CHANGES_REQUESTED** [review]
nice work, however, could you please try to extract most of config var in .env instead of statically type it .like for example database, host.., like that variables are going to change from time to time so you don't have to be specific for them.

File: src/models/user.js (hunk 1 above)
Comment (habinezadalvan) [reply src/models/user.js]: About constraints, as a team, I remember that we agreed on that whoever going to need User model will be the one to mention all the constraints, columns necessary. For now, the User model is a dummy just for confirming that the tables are being created in the database whenever we run the migration. Thank you

**Review by nkalyesubula — CHANGES_REQUESTED** [review]
Thanks, Christian. I suggest though that you change the port to 5432 every time you push as leaving it as 5433 will cause issues when running tests on Travis. Also, consider creating global variables in the .env file for the database name, and host

**Review by minega25 — APPROVED** [review]

**Review by nkalyesubula — APPROVED** [review]

File: .babelrc [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+    "presets": ["@babel/preset-env"],
+    "plugins": [["@babel/transform-runtime"]]
+ }
```
Comment (Mnickii) [inline .babelrc]: Add a new line

File: .sequelizerc [hunk 3]
```diff
@@ -0,0 +1,8 @@
+var path = require('path')
+
+module.exports = {
+  'config':          path.resolve('src', 'config', 'config.js'),
+  'migrations-path': path.resolve('src', 'migrations'),
+  'models-path':     path.resolve('src', 'models'),
+  'seeders-path':    path.resolve('src', 'seeders'),
+}
```
Comment (Mnickii) [inline .sequelizerc]: Add a newline

File: src/config/config.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  production: {
+    username: DB_USER,
+    password: DB_PASSWORD,
+    database: DB_NAME,
+    host: HOST,
+    dialect: 'postgres',
+    port: DB_PORT,
+    seederStorage: 'sequelize'
+  }
+};
```
Comment (Mnickii) [inline src/config/config.js]: Add a newline

File: src/models/user.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+    email: DataTypes.STRING,
+    password: DataTypes.STRING
+  }, {});
+  User.associate = function(models) {
+    // associations can be defined here
+  };
+  return User;
+};
```
Comment (Mnickii) [inline src/models/user.js]: Add new line

**Review by salviosage — APPROVED** [review]

File: .sequelizerc (hunk 3 above)
Comment (habinezadalvan) [reply .sequelizerc]: Done

File: .babelrc (hunk 2 above)
Comment (habinezadalvan) [reply .babelrc]: Done

File: src/config/config.js (hunk 4 above)
Comment (habinezadalvan) [reply src/config/config.js]: Done

**Pull request merged**